### ISSUE TITLE
[BUG]: Warnings are prioritized over errors

### ISSUE THREAD

**Issue by alvaromontoro** [jackdomleo7/Checka11y.css#151]
# Bug Report

## Details of the bug

Working on the `longdesc` task (issue #108), I found that in some cases where I expected an error, a more generic warning was displayed. This happens because the selectors for error and warning have the same specificity and in checka11y.scss, we have the following code:

```scss
/*! Checka11y.css v2.0.0 | MIT License | github.com/jackdomleo7/Checka11y.css */
@use "./errors/checka11y-errors.scss";
@use "./warnings/checka11y-warnings.scss";
```
If specificity is the same for both selectors, the one appearing later in the code will prevail. In this case the warning. Ideally, we should swap the position of the `@use` in the file above.

Checka11y.css version: 2.1.0

## What is the expected result?

If an element has multiple issues, errors should take precedence over warnings. 

## What is the actual result?

At equal specificity, the warning is displayed instead of the error.

## Acceptance criteria

- [ ] If an element has a warning and an error with equal specificity, errors should prevail

## Why is fixing this bug important?

People may disregard warnings as "not priorities" and ignore them, but those warnings may be hiding an error behind them. The most restrictive message should take precedence, so if an element has an error and a warning with the same specificity, the error should be displayed. (error > warning)

**Comment by alvaromontoro** [comment]
IF this is considered a valid bug, I could make the change as part of #108 (as it would have a dependency on it anyway) or put that one on hold until someone does this one.

**Comment by jackdomleo7** [comment]
Really good find @alvaromontoro! Honestly surprised I never spotted this. I'm happy with your solution (but could you add a comment explaining why warnings must be specified first) and I'm happy for you to add it to #108 (will just need to link both issues to your PR) 🙂

**Comment by alvaromontoro** [comment]
The idea behind this change is that if an element has more than one alert, the most severe one should be displayed. It's a worst-case scenario type of situation. If a component is flagged with a warning and an error, the latter should take priority because it is more important to fix. Warnings are "should fix," errors are "must fix."

How does this impact Checka11y.css? It's about specificity and the CSS cascade. If an element matches multiple rules, the properties from the selectors with the highest specificity are applied. If different selectors have the same specificity, the one that appears later in the code is used.

The checka11y.css file is generated by importing the errors and then the warnings:

```scss
@use "./errors/checka11y-errors.scss";
@use "./warnings/checka11y-warnings.scss";
```

There could be a case in which an element matches selectors for errors and warnings. If that's the case, and they have equal specificity, the warning ones will prevail because they appear later in the code. To solve this, we should reverse the order of the `@use` like this:

```scss
@use "./warnings/checka11y-warnings.scss";
@use "./errors/checka11y-errors.scss";
```

---

Let's see a practical example:

- If an image has a `longdesc` attribute, a warning should be displayed (because `longdesc` is deprecated and has known accessibility issues.)
- If an image has a `longdesc` attribute pointing to an image, an error should be displayed (as the linked image won't have an alt and will be inaccessible.)

The selector for the warning is `img[longdesc]`, and the selector for the error is `img[longdesc$=.jpg]`. Both have the same specificity (011).

Now imagine the image `<img src="..." alt="..." longdesc="desc.jpg" />`. It matches both selectors, and because they both have the same specificity, the one that appears last is the one that will be displayed. Checka11y.css will display the warning because the CSS file is generated by importing errors first and warnings second.
